### PR TITLE
Store persistently`WorkPreference`s locally in `Room`

### PR DESCRIPTION
**Pull request by Lynoxe**
I created a `Room` entity for the `WorkPreference` class. It is now stored persistently across app sessions and correctly synchronised with the `View`.

Regarding the `View` aspect, the main changes were made to `ReadingControlState`. It had now to use the `UseCase`s.

### CODE REVIEW

File: app/src/androidTest/java/com/github/leuludyha/ibdb/presentation/components/reading_list/ReadingStateControlTest.kt [hunk 1]
```diff
@@ -87,21 +78,24 @@ class ReadingStateControlTest {
         readingStateController.assertDoesNotExist()
     }
 
+    @OptIn(ExperimentalCoroutinesApi::class)
     @Test
-    fun likeAndDislikeModifiesEntryToPreferences() {
+    fun dislikeAndLikeModifiesEntryToPreferences() {
```
Comment (WollfieGitHub) [inline app/src/androidTest/java/com/github/leuludyha/ibdb/presentation/components/reading_list/ReadingStateControlTest.kt:83]: Ok my bad

File: app/src/main/java/com/github/leuludyha/ibdb/presentation/components/books/reading_list/controls/ReadingStateControlViewModel.kt [hunk 2]
```diff
@@ -1,13 +1,40 @@
 package com.github.leuludyha.ibdb.presentation.components.books.reading_list.controls
```
Comment (WollfieGitHub) [inline app/src/main/java/com/github/leuludyha/ibdb/presentation/components/books/reading_list/controls/ReadingStateControlViewModel.kt:1]: Can we just group all actions (save/load/delete) in a separate viewModel or useCase or whatever that we can then reuse with @Inject in other viewModels ?

**Review by WollfieGitHub — CHANGES_REQUESTED** [review]
Is there a way we could use something more clean as flows ? :( 
Also please address my concern about merging saving/delete/load-locally methods

File: app/src/main/java/com/github/leuludyha/ibdb/presentation/components/books/reading_list/controls/ReadingStateControlViewModel.kt (hunk 2 above)
Comment (Lynoxe) [reply app/src/main/java/com/github/leuludyha/ibdb/presentation/components/books/reading_list/controls/ReadingStateControlViewModel.kt:1]: Sure, you think having a `LocalUseCases` class grouping all of them would be better?

**Comment by Lynoxe** [comment]
> Is there a way we could use something more clean as flows ? :(
> Also please address my concern about merging saving/delete/load-locally methods

Honestly, from all the most recent official recommended tutorials I've seen, they all do it this way: providing a `Flow` in the `ViewModel` and calling `collectAsState()` inside the View. I don't think it is so dirty, the whole view is automatically updated whenever a change is made to the flow's value without any additional work from us. Moreover I think it is quite few lines of code (only calling the `UseCase` inside the `ViewModel` and `collectAsState` inside the View)

**Comment by WollfieGitHub** [comment]
Sorry I just merged a PR :(

File: app/src/main/java/com/github/leuludyha/ibdb/presentation/components/books/reading_list/controls/ReadingStateControlViewModel.kt (hunk 2 above)
Comment (WollfieGitHub) [reply app/src/main/java/com/github/leuludyha/ibdb/presentation/components/books/reading_list/controls/ReadingStateControlViewModel.kt:1]: Maybe something like `LocalRepository` which holds the load, delete and save method would be more suitable but yes, the idea is to regroup them

Comment (Lynoxe) [reply app/src/main/java/com/github/leuludyha/ibdb/presentation/components/books/reading_list/controls/ReadingStateControlViewModel.kt:1]: Then why not passing directly the `LibraryRepository`? How I see things is: either you want to be clear and restrictive on what functionalities you're using and use the `UseCase`s, or you want to be general and have access to a lot of methods and use `LibraryRepository`.
I agree that conceptually it is better to restrain our api to the necessary (hence having only the local functions of the Repository) but then I think we can simply use the `UseCase`s

Comment (WollfieGitHub) [reply app/src/main/java/com/github/leuludyha/ibdb/presentation/components/books/reading_list/controls/ReadingStateControlViewModel.kt:1]: Right, I was confused then, nevermind

**Review by WollfieGitHub — APPROVED** [review]
Looks good

**Pull request merged**